python/cudf/cudf/core/dataframe.py

```diff
@@ -5679,11 +5679,17 @@ def to_arrow(self, preserve_index=True):
 
         out = super(DataFrame, data).to_arrow()
         metadata = pa.pandas_compat.construct_metadata(
-            self,
-            out.schema.names,
-            [self.index],
-            index_descr,
-            preserve_index,
+            columns_to_convert=[
+                col.to_pandas()
```

Comment:
+1

Reply:
Removed the `to_pandas` here as we already error at the top for categoricals.